var/spack/repos/builtin/packages/tandem/package.py

```diff
@@ -58,10 +59,13 @@ class Tandem(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("eigen@3.4.0")
 
     depends_on("zlib-api")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack +knl", when="target=skylake:")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32 +cuda", when="+cuda")
-    depends_on("petsc@3.14.6:3.18.5 +int64 +mumps +scalapack memalign=32 +rocm", when="+rocm")
+    depends_on("petsc@3.14.6:3.21.2 +int64 +mumps +scalapack memalign=32")
+    depends_on("petsc@3.14.6:3.21.2 +int64 +mumps +scalapack +knl", when="target=skylake:")
+    depends_on("petsc@3.14.6:3.21.2 +int64 +mumps +scalapack memalign=32 +cuda", when="+cuda")
+    depends_on("petsc@3.14.6:3.21.2 +int64 +mumps +scalapack memalign=32 +rocm", when="+rocm")
+
+    depends_on("python@3", type="build", when="+python")
+    depends_on("py-numpy", type="build", when="+python")
```

Comment:
Can you explain the "build" dependency on python?

Reply:
This is because tandem relies on yateto https://github.com/SeisSol/yateto for code generation of its high performance kernels:
https://github.com/TEAR-ERC/tandem/blob/b70a134ffe729571e71bdc1fd9dde20ff3e2e7fd/app/CMakeLists.txt#L91